schema_utils.py

```diff
@@ -627,3 +627,23 @@ def is_valid_audio_language_code(obj):
             bool. Whether the given object is a valid audio language code.
         """
         return utils.is_valid_language_code(obj)
+
+    @staticmethod
+    def is_valid_thread_id(obj):
+        # type: (Text) -> bool
+        """Checks if the given obj (a string) represents a valid thread id.
+        The id has the form [entity_type].[entity_id].[generated_string]
```

Comment:
Can you use regex to validate the other parts too?

Reply:
Thinking about it it might be nice to provide a native support for regexes in the schema utils.

Reply:
Done, Please take a look.

Reply:
Not done? I was expecting some way you can put the regex directly in the schema so that you don't need to create a method for this.

Reply:
Yep I'm fine with that.

Reply:
Done